src/dlmalloc.rs

```diff
@@ -24,42 +25,22 @@ pub struct Dlmalloc {
     trim_check: usize,
     least_addr: *mut u8,
     release_checks: usize,
+    phantom: PhantomData<S>,
 }
 
-unsafe impl Send for Dlmalloc {}
-
-pub const DLMALLOC_INIT: Dlmalloc = Dlmalloc {
-    smallmap: 0,
-    treemap: 0,
-    smallbins: [0 as *mut _; (NSMALLBINS + 1) * 2],
-    treebins: [0 as *mut _; NTREEBINS],
-    dvsize: 0,
-    topsize: 0,
-    dv: 0 as *mut _,
-    top: 0 as *mut _,
-    footprint: 0,
-    max_footprint: 0,
-    seg: Segment {
-        base: 0 as *mut _,
-        size: 0,
-        next: 0 as *mut _,
-        flags: 0,
-    },
-    trim_check: 0,
-    least_addr: 0 as *mut _,
-    release_checks: 0,
-};
+unsafe impl<S: System + Send> Send for Dlmalloc<S> {}
```

Comment:
Probably should just make `trait System: Send`.

Reply:
`System` doesn't need to be `Send`, `Dlmalloc` needs to be.

Reply:
System is only used in Dlmalloc though? How is a non-Send `System` useful?